gocourse05/zooobservation/animal/animal_test.go

```diff
@@ -0,0 +1,70 @@
+package animal
+
+import (
+	"testing"
+
+	"zooobservation/camera"
+	"zooobservation/common"
+)
+
+func TestAnimal_Move(t *testing.T) {
+	direct := common.Left
+	var historyItems []common.HistoryItem
+
+	tiger := Animal{
+		Id: 1,
+		Camera: &camera.DayCamera{
+			Screenshot: "day_screenshot.png",
+		},
+		Species: "tiger",
+	}
+
+	history, err := tiger.Move(direct, historyItems)
+	if err != nil {
+		t.Fatalf("Error saving history to server with Move() Tiger: %v", err)
+	} else if history[0].Direction != direct || history[0].AnimalID != tiger.Id {
+		t.Errorf("We got direction=%s and animalID=%d instead of: %s, %d", history[0].Direction,
+			history[0].AnimalID, direct, tiger.Id)
+	}
```

Comment:
Так легше читається:
```suggestion
	if err != nil {
		t.Fatalf("Unexpected error: got=%v, but want=<nil>", err)
	}
	if len(history) == 0 {
		t.Fatalf("Unexpected empty history")
	}
	if history[0].AnimalID != tiger.Id {
		t.Errorf("Wrong history animalID: got=%v, but want=%v", history[0].AnimalID, tiger.Id)
	}
	if history[0].Direction != direct {
		t.Errorf("Wrong direction: got=%v, but want=%v", history[0].Direction, direct)
	}
```

Чому:

1. Один if - одна перевірка.
2. Захистились від ситуації, коли `history[0]` панікує.
3. Немає if-else. Так, ця конструкція погіршує читабельність.
4. Може зламатись `AnimalID` або `direct` - тоді в лозі тесту буде дві помилки замість однієї.
5. Поправили меседжі згідно з рекомендаціями https://go.dev/wiki/CodeReviewComments#useful-test-failures. Трохи суб'єктивно

Reply:
Пропоную редагувати схожим чином інші тести.